docker-app/qfieldcloud/core/management/commands/extractstoragemetadata.py

```diff
@@ -0,0 +1,133 @@
+import csv
+import logging
+import sys
+from typing import Generator, NamedTuple
+
+import boto3
+import mypy_boto3_s3
+from django.core.management.base import BaseCommand, CommandParser
+from qfieldcloud.core import utils
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Save metadata from S3 storage project files to disk."""
+
+    def add_arguments(self, parser: CommandParser):
+        parser.add_argument(
+            "-o", "--output", type=str, help="Optional: Name of output file"
+        )
+        parser.add_argument(
+            "-s3",
+            "--s3_credentials",
+            type=str,
+            help="""
+            Optional: S3 credentials as comma-separated key=value pairs.
+            Example: -s3 STORAGE_ACCESS_KEY_ID=...,STORAGE_SECRET_ACCESS_KEY=...,...
+        """,
+        )
+
+    def handle(self, *args, **options):
+        config = None
+
+        if options.get("s3_credentials"):
+            config = self.from_shell(options["s3_credentials"])
+            bucket = self.get_s3_bucket(config)
+
+        else:
+            bucket = utils.get_s3_bucket()
+
+        output_name = options.get("output")
+
+        if output_name:
+            name_suffix = output_name.rsplit(".", 1)
+
+            if len(name_suffix) != 2 or name_suffix[1] != "csv":
+                logger.error("The name of the output file should end with '.csv'")
+                sys.exit(1)
+        else:
+            output_name = "s3_storage_files.csv"
+
+        fields = (
+            "id",
+            "key",
+            "e_tag",
+            "size",
+            "last_modified",
+        )
+        rows = self.read_bucket_files(bucket, fields)
+
+        with open(output_name, "w") as fh:
+            writer = csv.writer(fh, delimiter=",")
+            writer.writerow(fields)
+            writer.writerows(rows)
+
+        logger.info(f"Successfully exported data to {output_name}")
+
+    class S3ConfigObject(NamedTuple):
+        STORAGE_ACCESS_KEY_ID: str
+        STORAGE_BUCKET_NAME: str
+        STORAGE_ENDPOINT_URL: str
+        STORAGE_REGION_NAME: str
+        STORAGE_SECRET_ACCESS_KEY: str
```

Comment:
If you insist of keeping this, please move it to the begining of the class. I would prefer to drop it and just use a simple dict. No strong opinion though.